FWCore/ParameterSet/src/defaultModuleLabel.cc

```diff
@@ -7,6 +7,9 @@ namespace edm {
     // remove all colons (module type may contain namespace)
     label.erase(std::remove(label.begin(), label.end(), ':'), label.end());
 
+    // remove everything after first '@' (used for module alternatives)
+    label.erase(std::find(label.begin(), label.end(), '@'), label.end());
```

Comment:
It would probably be a good idea to make `@` a named constant (and use the constant also elsewhere)

Reply:
After discussing with @Dr15Jones we concluded the code here is ok (`@` is used as special character in many other places in the framework as well, and its uses are somewhat findable with grep)